roles/v2/mining-proxy/src/lib/upstream_mining.rs

```diff
@@ -35,6 +38,105 @@ pub type StdFrame = StandardSv2Frame<Message>;
 pub type EitherFrame = StandardEitherFrame<Message>;
 pub type ProxyRemoteSelector = Prs<DownstreamMiningNode>;
 
+use std::sync::atomic::AtomicBool;
+/// USED to make sure that if a future new_temnplate and a set_new_prev_hash are received together
+/// the future new_temnplate is always handled before the set new prev hash.
+pub static IS_NEW_TEMPLATE_HANDLED: AtomicBool = AtomicBool::new(true);
```

Comment:
Instead of being static, should this be an `Arc<AtomicBool>` or similar in a field in a struct somewhere?

Reply:
Why do you think that would be better to have it in a field in a struct somewhere? Maybe it would be a little bit more safe but not so much cause the visibility need to be almost global. We could initialize it in main and pass it around that would make the codebase more complex not sure if it is worth it.